### PR TITLE
[JENKINS-33947] - Fix keyboard navigation on setup wizard

### PR DESCRIPTION
**Pull request by kzantow**
Keyboard navigation was broken a number of ways, focus was lost and it was impossible to click 'dependencies'. This PR addresses those issues.

Fixing: https://issues.jenkins-ci.org/browse/JENKINS-33947

### CODE REVIEW

File: war/src/main/js/templates/pluginSelectionPanel.hbs [hunk 1]
```diff
@@ -31,17 +31,17 @@
...
-							{{../../../translations.installWizard_installIncomplete_dependenciesLabel}}<span class="badge">{{dependencyCount ../plugin.name}}</span>
-						</div>
-					{{/hasDependencies}}
 					<label>
 						<span class="title">
 							<input type="checkbox" id="chk-{{plugin.name}}" name="{{plugin.name}}" value="{{searchTerm}}" {{#inSelectedPlugins plugin.name}}checked="checked"{{/inSelectedPlugins}}/>
 							{{plugin.title}}
 							<a href="{{plugin.website}}" target="_blank">{{../../translations.installWizard_websiteLinkLabel}}</a>
 						</span>
+						{{#hasDependencies plugin.name}}
```
Comment (kzantow) [inline war/src/main/js/templates/pluginSelectionPanel.hbs]: Moved to make tab order more logical

File: war/src/main/less/pluginSetupWizard.less [hunk 2]
```diff
@@ -349,7 +349,7 @@
 					&.selected label {
 						background: #dff0d8;
 						margin: 4px -7px;
-						padding: 6px 0 6px 26px;
+						padding: 6px 1px 6px 26px;
```
Comment (kzantow) [inline war/src/main/less/pluginSetupWizard.less]: Unrelated minor issue: was causing text to shift when plugins were selected

**Comment by kzantow** [comment]
@reviewbybees

File: war/src/main/js/pluginSetupWizardGui.js [hunk 3]
```diff
@@ -200,6 +209,19 @@ var createPluginSetupWizard = function(appendTarget) {
...
+			try {
+				var e = $('body')[0];
+				for (var i = focusPath.length-1; i >= 0; i--) {
+					e = e.children[focusPath[i]];
+				}
+				if (document.activeElement != e) {
+					e.focus();
+				}
+			} catch (e) {
+				console.log('Unable to restore focus...');
```
Comment (recena) [inline war/src/main/js/pluginSetupWizardGui.js]: It is not a good practice to use this in production environment.

**Comment by recena** [comment]
🐜  IMO we should follow the W3C ARIA recommendations.

**Comment by kzantow** [comment]
@recena goal here was to fix keyboard navigation, not necessarily all the accessibility issues (there are still a number of them)... but what ARIA recommendation(s) are you referring to for this, specifically?

**Comment by daniel-beck** [comment]
> `src/main/js/pluginSetupWizardGui.js: line 235, col 23, Value of 'e' may be overwritten in IE 8 and earlier.`

@kzantow Could be legitimate. Master doesn't fail to build this way.

**Comment by kzantow** [comment]
@daniel-beck yes, could be legitimate. I'm pretty sure we don't support IE8 anymore, though (so we should modify any lint rules to exclude this check).

**Comment by recena** [comment]
@kzantow Easier if we solve this Lint error, no? 😄

**Comment by daniel-beck** [comment]
@kzantow Either way, this PR isn't mergeable as is.

**Comment by oleg-nenashev** [comment]
🐝 since I do not see obvious issues in the code. Manual testing confirms it works at least.
BTW, I still vote for creation of a follow-up EPIC, which would introduce some test coverage in the Setup Wizard

**Comment by oleg-nenashev** [comment]
@recena Are you fine with the current version?

**Comment by recena** [comment]
:+1: but I agree with the creation of follow-up

**Comment by oleg-nenashev** [comment]
I suppose @reviewbybees is done
Waiting for the feedback from @jenkinsci/code-reviewers

**Pull request merged**